internal/provider/resource_tfe_organization_run_task.go

```diff
@@ -1,209 +1,272 @@
+	resp.Schema = schema.Schema{
+		Version: 0,
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed:    true,
+				Description: "Service-generated identifier for the task",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"name": schema.StringAttribute{
 				Required: true,
 			},
-
-			"organization": {
-				Type:     schema.TypeString,
+			"organization": schema.StringAttribute{
 				Optional: true,
 				Computed: true,
-				ForceNew: true,
+				// From ForceNew: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
 			},
-
-			"url": {
-				Type:         schema.TypeString,
-				Required:     true,
-				ValidateFunc: validation.IsURLWithHTTPorHTTPS,
+			"url": schema.StringAttribute{
+				Required: true,
+				Validators: []validator.String{
+					customValidators.IsURLWithHTTPorHTTPS(),
+				},
 			},
-
-			"category": {
-				Type:     schema.TypeString,
-				Default:  "task",
+			"category": schema.StringAttribute{
 				Optional: true,
+				Computed: true,
+				Default:  stringdefault.StaticString("task"),
 			},
-
-			"hmac_key": {
-				Type:      schema.TypeString,
+			"hmac_key": schema.StringAttribute{
 				Sensitive: true,
-				Default:   "",
 				Optional:  true,
+				Computed:  true,
+				Default:   stringdefault.StaticString(""),
 			},
-
-			"enabled": {
-				Type:     schema.TypeBool,
-				Default:  true,
+			"enabled": schema.BoolAttribute{
 				Optional: true,
+				Computed: true,
+				Default:  booldefault.StaticBool(true),
 			},
-
-			"description": {
-				Type:     schema.TypeString,
+			"description": schema.StringAttribute{
 				Optional: true,
+				Computed: true,
+				Default:  stringdefault.StaticString(""),
 			},
 		},
 	}
 }
 
-func resourceTFEOrganizationRunTaskCreate(d *schema.ResourceData, meta interface{}) error {
-	config := meta.(ConfiguredClient)
+func (r *resourceOrgRunTask) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state modelTFEOrganizationRunTaskV0
 
-	// Get the task name and organization.
-	name := d.Get("name").(string)
-	organization, err := config.schemaOrDefaultOrganization(d)
-	if err != nil {
-		return err
+	// Read Terraform current state into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
 	}
 
-	// Create a new options struct.
-	options := tfe.RunTaskCreateOptions{
-		Name:        name,
-		URL:         d.Get("url").(string),
-		Category:    d.Get("category").(string),
-		HMACKey:     tfe.String(d.Get("hmac_key").(string)),
-		Enabled:     tfe.Bool(d.Get("enabled").(bool)),
-		Description: tfe.String(d.Get("description").(string)),
+	var organization string
+	resp.Diagnostics.Append(r.config.dataOrDefaultOrganization(ctx, req.State, &organization)...)
+
+	if resp.Diagnostics.HasError() {
+		return
 	}
```

Comment:
I don't think the organization is used or needed for Read. These lines can just be deleted since `modelFromTFEOrganizationRunTask` gets the organization from the API

Reply:
Agreed. Removed.